tests/countries/test_united_states.py

```diff
@@ -469,17 +513,95 @@ def test_columbus_day_states(self):
             "2022-10-10",
             "2023-10-09",
         )
-
-        for subdiv in ("AK", "AR", "DE", "FL", "HI", "NV"):
-            self.assertNoHoliday(self.state_hols[subdiv], dt)
-            self.assertNoHolidayName(name, self.state_hols[subdiv])
-
-        self.assertHolidayName("Native American Day", self.state_hols["SD"], dt)
+        for subdiv, change_year in (
+            ("AK", 2015),
+            ("DC", 2019),
+            ("ME", 2019),
+            ("NM", 2019),
+            ("NE", 2020),
+            ("VA", 2020),
+        ):
+            self.assertNoHolidayName(name_1, self.state_hols[subdiv], range(1865, 1970))
+            self.assertHolidayName(name_1, self.state_hols[subdiv], range(1971, change_year))
+            self.assertHolidayName(name_2, self.state_hols[subdiv], range(change_year, 2050))
+            self.assertHoliday(self.state_hols[subdiv], dt)
 
         self.assertHolidayName(
             "Columbus Day and Puerto Rico Friendship Day", self.state_hols["VI"], dt
         )
 
+    def test_columbus_day_al(self):
+        name_1 = "Columbus Day / Fraternal Day"
+        name_2 = "Columbus Day / American Indian Heritage Day / Fraternal Day"
+        dt = (
+            "2010-10-11",
+            "2011-10-10",
+            "2012-10-08",
+            "2013-10-14",
+            "2014-10-13",
+            "2015-10-12",
+            "2016-10-10",
+            "2017-10-09",
+            "2018-10-08",
+            "2019-10-14",
+            "2020-10-12",
+            "2021-10-11",
+            "2022-10-10",
+            "2023-10-09",
+        )
+        self.assertNoHolidayName(name_1, self.state_hols["AL"], range(1865, 1970))
+        self.assertHolidayName(name_1, self.state_hols["AL"], range(1971, 2000))
+        self.assertHolidayName(name_2, self.state_hols["AL"], range(2000, 2050))
+        self.assertHoliday(self.state_hols["AL"], dt)
+
+    def test_columbus_day_ri(self):
```

Comment:
Any source on Rhode Island ones?

Reply:
https://web.archive.org/web/20230000000000*/https://www.sos.ri.gov/divisions/civics-and-education/reference-desk/ri-state-holidays